kernel/src/upcall.rs

```diff
@@ -15,109 +15,78 @@ use crate::ErrorCode;
 #[derive(Copy, Clone, PartialEq, Debug)]
 pub struct UpcallId {
     pub driver_num: usize,
-    pub subscribe_num: usize,
+    pub subscribe_num: usize, //TODO: use u32?
```

Comment:
In #2462 these were ported to u32

Reply:
yeah... `usize` if they are register sized and `u32` if they are 4-byte words (which happen to be the same on all of our platforms). Since an upcall is something that models the interaction over a system call boundary, it seems to me it should be a `usize`. E.g., "how many subscribe numbers can there be?" answer: "the max value that can fit in a CPU register", not, "2^32".